aws/resource_aws_dms_event_subscription.go

```diff
@@ -0,0 +1,251 @@
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"event_categories": {
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"sns_topic_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"source_ids": {
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+				ForceNew: true,
+				Optional: true,
+			},
+			"source_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				// The API suppors modification but doing so loses all source_ids
+				ForceNew: true,
+			},
+			"tags": {
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAwsDmsEventSubscriptionCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dmsconn
+
+	request := &dms.CreateEventSubscriptionInput{
+		Enabled:          aws.Bool(d.Get("enabled").(bool)),
+		SnsTopicArn:      aws.String(d.Get("sns_topic_arn").(string)),
+		SubscriptionName: aws.String(d.Get("name").(string)),
+		SourceType:       aws.String(d.Get("source_type").(string)),
+		Tags:             dmsTagsFromMap(d.Get("tags").(map[string]interface{})),
+	}
+
+	if v, ok := d.GetOk("event_categories"); ok {
+		request.EventCategories = expandStringList(v.(*schema.Set).List())
+	}
+
+	if v, ok := d.GetOk("source_ids"); ok {
+		request.SourceIds = expandStringList(v.(*schema.Set).List())
+	}
+
+	_, err := conn.CreateEventSubscription(request)
+
+	if err != nil {
+		return fmt.Errorf("Error creating DMS event subscription: %s", err)
+	}
+
+	d.SetId(d.Get("name").(string))
+
+	log.Println("[DEBUG] DMS create event subscription", request)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"creating", "modifying"},
+		Target:     []string{"active"},
+		Refresh:    resourceAwsDmsEventSubscriptionStateRefreshFunc(conn, d.Id()),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		MinTimeout: 10 * time.Second,
+		Delay:      10 * time.Second,
+	}
+
+	// Wait, catching any errors
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("error waiting for DMS event subscription (%s) creation: %s", d.Id(), err)
+	}
+
+	return resourceAwsDmsEventSubscriptionUpdate(d, meta)
+}
+
+func resourceAwsDmsEventSubscriptionUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dmsconn
+
+	request := &dms.ModifyEventSubscriptionInput{
```

Comment:
To support tags updates and this API call separately, the existing update code will need to be wrapped in:

```go
if d.HasChanges("enabled", "event_categories", "sns_topic_arn", "source_type") {
  // ... existing update code ...
}
```